system/ee/ExpressionEngine/Service/Generator/ExtensionHookGenerator.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * This source file is part of the open source project
+ * ExpressionEngine (https://expressionengine.com)
+ *
+ * @link      https://expressionengine.com/
+ * @copyright Copyright (c) 2003-2022, Packet Tide, LLC (https://www.packettide.com)
+ * @license   https://expressionengine.com/license Licensed under Apache License, Version 2.0
+ */
+
+namespace ExpressionEngine\Service\Generator;
+
+use ExpressionEngine\Library\Filesystem\Filesystem;
+use ExpressionEngine\Library\String\Str;
+use ExpressionEngine\Service\Generator\Enums\Hooks;
+
+class ExtensionHookGenerator
+{
+    public $name;
+    public $addon;
+    protected $filesystem;
+    protected $str;
+    protected $generatorPath;
+    protected $addonPath;
+    protected $stubPath;
+
+    public function __construct(Filesystem $filesystem, Str $str, array $data)
+    {
+        // Set FS
+        $this->filesystem = $filesystem;
+        $this->str = $str;
+
+        // Set required data for generator to use
+        $this->name = $this->str->snakecase($data['name']);
+        $this->ExtensionHookName = $this->str->studly($this->name);
+        $this->addon = $this->str->snakecase($data['addon']);
+
+        // Set up addon path, generator path, and stub path
+        $this->init();
+
+        $addonSetupArray = require $this->addonPath . 'addon.setup.php';
+        $this->namespace = $addonSetupArray['namespace'];
+    }
+
+    private function init()
+    {
+        $this->generatorPath = SYSPATH . 'ee/ExpressionEngine/Service/Generator';
+        $this->addonPath = SYSPATH . 'user/addons/' . $this->addon . '/';
+
+        // Make sure the addon exists
+        if (! ee('Addon')->get($this->addon)) {
+            throw new \Exception(lang('cli_error_the_specified_addon_does_not_exist'), 1);
+        }
+
+        // Get stub path
+        $this->stubPath = $this->generatorPath . '/stubs/MakeAddon/Extension/';
+    }
+
+    public function build()
+    {
+        $hookData = Hooks::getByKey(trim(strtoupper($this->name)));
+
+        // If we didnt get a real hook, set up a default
+        if ($hookData === false) {
+            $hookData = [
+                'name' => $hook,
```

Comment:
This is undefined, but not sure where this should be passed from

Reply:
Good catch 👍